### PR TITLE
DOCS-339: MinIO ILM Migration + Tiering

### PR DESCRIPTION
**Pull request by ravindk89**
# Summary

Create and update reference documentation + tutorials for MinIO ILM + Remote Tiers

# Goals
- Refresh `mc ilm` for missed updates
- Add `mc admin remote` command
- Add tutorials for transitioning objects to s3, gcs, azure
- Add tutorial for object expiration
- Create landing page for object lifecycle management

# Non-Goals

- Complete cleanup of reference pages

# Build Instructions

The MinIO docs require Python 3.9+ and Node for building. For Node, use v14.5.0 or later.

1) Clone the repo, Checkout the branch (`git checkout -B installation-fixes origin/DOCS-339`)
2) Run `npm install` from the branch
3) Create a python venv (`python -m venv venv`). You can skip this step if it already exists
4) Activate the venv (`source venv/bin/activate`)
5) Install requirements (`pip install -r requirements.txt`)
6) Run `make html`
7) Run `python -m http.server --directory build/DOCS-339/html`
8) Open 127.0.0.1:8000 in your browser and take a poke around.

If you run into an error running `python -m http.server` related to `--directory`, you're python version is likely older. CD into the `build/<branch>/html` directory and run the command without the `--directory` argument.

Errors related to `wheel` are likely due to missing `wheel` + other OS-specific dependencies. See https://stackoverflow.com/a/59596814

# Key URLS

- http://127.0.0.1:8000/lifecycle-management/lifecycle-management-overview.html
- http://127.0.0.1:8000/lifecycle-management/create-lifecycle-management-expiration-rule.html
- http://127.0.0.1:8000/lifecycle-management/transition-objects-to-s3.html
- http://127.0.0.1:8000/lifecycle-management/transition-objects-to-gcs.html
- http://127.0.0.1:8000/lifecycle-management/transition-objects-to-azure.html
- http://127.0.0.1:8000/reference/minio-cli/minio-mc/mc-ilm.html#mc-ilm-add
- http://127.0.0.1:8000/reference/minio-cli/minio-mc-admin/mc-admin-tier.html

### CODE REVIEW

File: source/includes/common-minio-tiering.rst
Comment (ravindk89) [inline source/includes/common-minio-tiering.rst:3]: This chunk gets included in all three of the transition tutorials as part of the step for creating the new transition rules.

File: source/includes/common-minio-tiering.rst
Comment (ravindk89) [inline source/includes/common-minio-tiering.rst:73]: This chunk gets included in the step where users create users/policies.

File: source/includes/common-minio-tiering.rst
Comment (ravindk89) [inline source/includes/common-minio-tiering.rst:87]: This wont work until I merge to master :/ sorry!

Comment (ravindk89) [reply source/includes/common-minio-tiering.rst:87]: If you want to cheat, change the URL to the python server root instead

**Review by poornas — CHANGES_REQUESTED** [review]

File: source/lifecycle-management/lifecycle-management-overview.rst
Comment (krisis) [inline source/lifecycle-management/lifecycle-management-overview.rst:15]:
```suggestion
MinIO automatically moves the object to a configured remote storage
```

File: source/lifecycle-management/lifecycle-management-overview.rst
Comment (krisis) [inline source/lifecycle-management/lifecycle-management-overview.rst:22]: nit: typo
```suggestion
and conversion to or from XML may be required. 
```

File: source/lifecycle-management/lifecycle-management-overview.rst
Comment (krisis) [inline source/lifecycle-management/lifecycle-management-overview.rst:80]:
```suggestion
"expiration" involves performing a ``DELETE`` operation on the object. For 
```

File: source/lifecycle-management/transition-objects-to-azure.rst
Comment (krisis) [inline source/lifecycle-management/transition-objects-to-azure.rst:73]: nit: typo
```suggestion
Refer to the `Azure RBAC
```

File: source/lifecycle-management/lifecycle-management-overview.rst
Comment (krisis) [inline source/lifecycle-management/lifecycle-management-overview.rst:52]: Suggestion
```suggestion
data to a supported Cloud Service Provider object storage You can then use the
```

File: source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst
Comment (krisis) [inline source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst:181]:
```suggestion
      have permission to perform read/write/list/delete operations on the 
```

File: source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst
Comment (krisis) [inline source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst:203]:
```suggestion
      must have permission to perform read/write/list/delete operations on the
```

File: source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst
Comment (krisis) [inline source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst:227]:
```suggestion
      read/write/list/delete operations on the remote bucket or bucket prefix.
```

File: source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst
Comment (krisis) [inline source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst:312]:
```suggestion
      have permission to perform read/write/list/delete operations on the 
```

File: source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst
Comment (krisis) [inline source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst:336]:
```suggestion
      must have permission to perform read/write/list/delete operations on the
```

File: source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst
Comment (krisis) [inline source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst:360]:
```suggestion
      permission to perform read/write/list/delete operations on the remote bucket
```

File: source/reference/minio-cli/minio-mc/mc-ilm.rst
Comment (ravindk89) [reply source/reference/minio-cli/minio-mc/mc-ilm.rst:298]: Hm - that would make sense for transition-days, but not transition-date. I'm not sure what I was thinking here, so I'll remove this.

File: source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst
Comment (krisis) [inline source/reference/minio-cli/minio-mc-admin/mc-admin-tier.rst:330]:
```suggestion
   .. mc-cmd:: account-name
```

File: source/reference/minio-cli/minio-mc/mc-ilm.rst
Comment (ravindk89) [reply source/reference/minio-cli/minio-mc/mc-ilm.rst:298]: Hm - that would make sense for transition-days, but not transition-date. I'm not sure what I was thinking here, so I'll remove this.

File: source/lifecycle-management/transition-objects-to-gcs.rst
Comment (krisis) [inline source/lifecycle-management/transition-objects-to-gcs.rst:73]:
```suggestion
Refer to the `GCS IAM permissions
```

File: source/lifecycle-management/transition-objects-to-s3.rst
Comment (krisis) [inline source/lifecycle-management/transition-objects-to-s3.rst:82]:
```suggestion
Refer to the :aws-docs:`Amazon S3 Permissions 
```

**Review by krisis — APPROVED** [review]

**Review by poornas — APPROVED** [review]
LGTM

**Comment by ravindk89** [comment]
Holding off on merging until we have the latest release of `minio` + and the `mc` changes from the pending PR on `mc admin tier`.

**Pull request merged**
